group_project_v2/mixins.py

```diff
@@ -125,7 +126,180 @@ def real_user_id(self, anonymous_student_id):
         return self._known_real_user_ids[anonymous_student_id]
 
 
-class WorkgroupAwareXBlockMixin(UserAwareXBlockMixin, CourseAwareXBlockMixin):
+class SettingsMixin(object):
+
+    def _get_setting(self, setting, default):
+        result = default
+        settings_service = self.runtime.service(self, "settings")
+        if settings_service:
+            xblock_settings = settings_service.get_settings_bucket(self)
+            if xblock_settings and setting in xblock_settings:
+                result = xblock_settings[setting]
+        return result
+
+
+class AuthXBlockMixin(SettingsMixin, ProjectAPIXBlockMixin):
+
+    _DEFAULT_TA_ROLE = ('assistant',)
+
+    _SEE_DASHBOARD_ROLE_PERMS = "see_dashboard_role_perms"
```

Comment:
@jbzdak Maybe :) Basically, the question between `access_dashboard_groups` and access_dashboard_roles` is that do you want to reflect how they are used (i.e. roles), or what are they under the hood (groups). Basically Apros uses LMS groups as roles - so do you want to call them roles, or groups. I'll leave this final decision to you :)